src/test/java/com/amihaiemil/camel/StrictYamlSequenceTest.java

```diff
@@ -0,0 +1,118 @@
+        elements.add(new Scalar("key1"));
+        elements.add(new Scalar("key2"));
+        elements.add(new Scalar("key3"));
+        YamlSequence sequence = new StrictYamlSequence(new RtYamlSequence(elements));
+
+        MatcherAssert.assertThat(
+            sequence.children(), Matchers.not(Matchers.emptyIterable())
+        );
+    }
+
+    /**
+     * StringYamlSequence can throw YamlNodeNotFoundException
+     * when the demanded YamlMapping is not found.
+     */
+    @Test (expected = YamlNodeNotFoundException.class)
+    public void exceptionOnNullMapping() {
+        YamlMapping origin = Mockito.mock(YamlMapping.class);
+        Mockito.when(origin.yamlMapping("key")).thenReturn(null);
+        YamlMapping strict = new StrictYamlMapping(origin);
+        strict.yamlMapping("key");
+    }
+
+    /**
+     * StringYamlSequence can throw YamlNodeNotFoundException
+     * when the demanded YamlSequence is not found.
+     */
+    @Test (expected = YamlNodeNotFoundException.class)
+    public void exceptionOnNullSequence() {
+        YamlMapping origin = Mockito.mock(YamlMapping.class);
+        Mockito.when(origin.yamlSequence("key")).thenReturn(null);
+        YamlMapping strict = new StrictYamlMapping(origin);
+        strict.yamlSequence("key");
+    }
+
+    /**
+     * StringYamlSequence can throw YamlNodeNotFoundException
+     * when the demanded String is not found.
+     */
+    @Test (expected = YamlNodeNotFoundException.class)
+    public void exceptionOnNullString() {
+        YamlSequence origin = Mockito.mock(YamlSequence.class);
+        Mockito.when(origin.string(1)).thenReturn(null);
+        YamlSequence strict = new StrictYamlSequence(origin);
+        strict.string(1);
+    }
+
+    /**
+     * StrictYamlSequence can return the number of elements in a sequence.
+     */
+    @Test
+    public void returnsSize() {
+        List<YamlNode> elements = new LinkedList<>();
+        elements.add(new Scalar("key1"));
+        elements.add(new Scalar("key2"));
+        YamlSequence sequence = new StrictYamlSequence(new RtYamlSequence(elements));
+
+        MatcherAssert.assertThat(sequence.children().size(), Matchers.equalTo(2));
+    }
+
+    /**
+     * StringYamlSequence can fetch a YamlMapping based in its key.
+     */
+    @Test
+    public void returnsMapping() {
+        YamlMapping origin = Mockito.mock(YamlMapping.class);
+        YamlMapping found = Mockito.mock(YamlMapping.class);
+        YamlNode key = new Scalar("key");
+        Mockito.when(origin.yamlMapping(key)).thenReturn(found);
+        YamlMapping strict = new StrictYamlMapping(origin);
+        MatcherAssert.assertThat(
+            strict.yamlMapping("key"), Matchers.equalTo(found)
+        );
+    }
+
+    /**
+     * StringYamlSequence can fetch a YamlSequence based in its key.
+     */
+    @Test
+    public void returnsSequence() {
+        YamlMapping origin = Mockito.mock(YamlMapping.class);
```

Comment:
@salikjan Same here, this unit test is from StrictYamlMapping. You should modify it :D